addons/mail/static/src/component_hooks/use_component_to_model/use_component_to_model.js

```diff
@@ -24,10 +24,10 @@ export function useComponentToModel({ fieldName, modelName, propNameAsRecordLoca
     }
     onWillUpdateProps(nextProps => {
         const currentRecord = modelManager.models[modelName].get(component.props[propNameAsRecordLocalId]);
-        if (currentRecord) {
+        const nextRecord = modelManager.models[modelName].get(nextProps[propNameAsRecordLocalId]);
+        if (currentRecord && currentRecord !== nextRecord) {
```

Comment:
If `currentRecord === nextRecord`, it seems useless to be calling the following update too. It's less bad because it's pretty much a useless update and not detrimental, but while we're at it we might as well optimize it fully.

Reply:
Or maybe not actually? It depends whether the component changed or not.

Reply:
Component could have changed, ref could have changed too (it depends on how OWL makes the reconciliation of vnodes). I'm pretty sure we should not trust any of that, so it's much safer to do a useless update than actually miss a genuine one.

Reply:
This is not a ref though it's the actual component,  `component.el` (or `.root` in v2) could change but it's not what it saved here.

But just the fact that the component itself could change is enough to invalidate my previous comments.